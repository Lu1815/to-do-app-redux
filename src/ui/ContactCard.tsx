interface IContactCardProps {
    contactName: string,
    contactImage: string
}

const ContactCard = ({ contactImage, contactName }: IContactCardProps) => {
    return (
        <div className="bg-white border-gray-200 border-2 py-8 px-10 text-center rounded-md shadow-lg w-full mx-auto">
            <div className="flex flex-row gap-3 justify-left mx-auto">
                <img className="w-20 h-20 object-cover rounded-full shadow-lg" src={contactImage} alt="Contact Avatar" />
                <div className="grid content-center">
                    <h2 className={"font-semibold text-left align-middle max-h-min text-2xl md:text-lg lg:text-2xl break-words"}>{contactName}</h2>
                </div>
            </div>
        </div>
    )
}

export default ContactCard