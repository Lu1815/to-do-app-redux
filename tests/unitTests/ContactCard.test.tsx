import React from 'react';
import { render, screen } from '@testing-library/react';
import ContactCard from '../../src/ui/ContactCard';

describe('ContactCard Component', () => {
  const mockContact = {
    contactName: 'John Doe',
    contactImage: 'https://example.com/avatar.jpg',
  };

  it('renders without crashing', () => {
    render(<ContactCard contactName={mockContact.contactName} contactImage={mockContact.contactImage} />);
    expect(screen.getByText('John Doe')).toBeInTheDocument();
    expect(screen.getByAltText('Contact Avatar')).toBeInTheDocument();
  });

  it('displays the correct contact name and image', () => {
    render(<ContactCard contactName={mockContact.contactName} contactImage={mockContact.contactImage} />);
    const contactNameElement = screen.getByText('John Doe');
    const contactImageElement = screen.getByAltText('Contact Avatar');

    expect(contactNameElement).toBeInTheDocument();
    expect(contactImageElement).toBeInTheDocument();
    expect(contactImageElement).toHaveAttribute('src', 'https://example.com/avatar.jpg');
  });

  it('applies proper styling', () => {
    render(<ContactCard contactName={mockContact.contactName} contactImage={mockContact.contactImage} />);
    const contactNameContainer = screen.getByText('John Doe').parentElement; 
    const contactCardContentContainer = contactNameContainer?.parentElement;
    const contactCardElement = contactCardContentContainer?.parentElement;

    expect(contactNameContainer).toHaveClass('grid');
    expect(contactNameContainer).toHaveClass('content-center');

    expect(contactCardContentContainer).toHaveClass('flex');
    expect(contactCardContentContainer).toHaveClass('flex-row');
    expect(contactCardContentContainer).toHaveClass('gap-3');
    expect(contactCardContentContainer).toHaveClass('justify-left');
    expect(contactCardContentContainer).toHaveClass('justify-left');
    expect(contactCardContentContainer).toHaveClass('mx-auto');

    expect(contactCardElement).toHaveClass('bg-white');
    expect(contactCardElement).toHaveClass('border-gray-200');
    expect(contactCardElement).toHaveClass('border-2');
    expect(contactCardElement).toHaveClass('py-8');
    expect(contactCardElement).toHaveClass('px-10');
    expect(contactCardElement).toHaveClass('text-center');
    expect(contactCardElement).toHaveClass('rounded-md');
    expect(contactCardElement).toHaveClass('shadow-lg');
    expect(contactCardElement).toHaveClass('w-full');
    expect(contactCardElement).toHaveClass('mx-auto');
  });

  it('matches snapshot', () => {
    const { container } = render(<ContactCard contactName={mockContact.contactName} contactImage={mockContact.contactImage} />);
    expect(container).toMatchSnapshot();
  });
});
