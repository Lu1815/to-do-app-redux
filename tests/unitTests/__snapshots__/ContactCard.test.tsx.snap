// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactCard Component matches snapshot 1`] = `
<div>
  <div
    class="bg-white border-gray-200 border-2 py-8 px-10 text-center rounded-md shadow-lg w-full mx-auto"
  >
    <div
      class="flex flex-row gap-3 justify-left mx-auto"
    >
      <img
        alt="Contact Avatar"
        class="w-20 h-20 object-cover rounded-full shadow-lg"
        src="https://example.com/avatar.jpg"
      />
      <div
        class="grid content-center"
      >
        <h2
          class="font-semibold text-left align-middle max-h-min text-2xl md:text-lg lg:text-2xl break-words"
        >
          John Doe
        </h2>
      </div>
    </div>
  </div>
</div>
`;
